pyso/stack_search_module.py

```diff
@@ -6,74 +6,49 @@
-        async with session.get(url, params=params) as resp:
-            return await resp.json()
-
-async def search(query):
-    """ 
-    Searches stack overflow for an answer to the provided query and returns a StackAnswer object containing the result.
-    query: Text to search SO for
-    """
-
-    async with aiohttp.ClientSession() as session:
-        # Find a question matching the query
-        question = await find_question(session, query)
-
-        # Grab the the first (top-voted) answer
-        answer_id = question.get("items", [{}])[0].get('accepted_answer_id', -1)
-
-        # Grab the answer's JSON
-        answer = None
-        async with aiohttp.ClientSession() as session_thing:
-            answer = await fetchAnswer(session_thing, answer_id)
+class pyso:
+    def __init__(self, access_token, key='uKDJS)Ck32rrSYrweVOoOQ(('):
+        """
+        Creates a PySO instance that allows you to search Stack Overflow.
         
-        # Convert the answer's JSON into a StackAnswer object
-        answer_dict = {
-                'link': answer.get('items', [{}])[0].get('share_link', "Not Found!"), 
-                'body': answer.get('items', [{}])[0].get('body_markdown', "Not Found!"), 
-                'answerer': answer.get('items', [{}])[0].get('owner', {}).get('display_name', "Not Found!"), 
-                'title': question.get('items', [{}])[0].get('title', "Not Found!"), 
-                'score': answer.get('items', [{}])[0].get('score', "Not Found!"),
-                'answerer_profile_image': answer.get('items', [{}])[0].get('owner', {}).get('profile_image', "https://cdn.discordapp.com/avatars/196989358165852160/b7645b3661eaf16bb2510c2292057890.png?size=256")
-        }
-
-        # Debug log if required
-        #print("Link => {}\nBody => {}\nUsername => {}\nTitle => {}\nScore => {}\n".format(answer['link'], answer['body'], answer['answerer'], answer['title'], answer['score']))
-
-        return answer_dict
+        access_token: string | your stack exchange access token
+        key: string | your application's key. Default has been set to the PySO app.
+        """
+
+        self.credentials = (key, access_token)
+
+    async def search(query):
+        """ 
+        Searches stack overflow for an answer to the provided query and returns a StackAnswer object containing the result.
+        query: string | text to search Stack Overflow for
+        """
+
+        async with aiohttp.ClientSession() as session:
+            # Find a question matching the query
+            question = await find_question(session, query)
+
+            # Grab the the first (top-voted) answer
+            answer_id = question.get("items", [{}])[0].get('accepted_answer_id', -1)
+
+            # Grab the answer's JSON
+            answer = None
+            async with aiohttp.ClientSession() as session_thing:
+                answer = await fetchAnswer(session_thing, answer_id)
```

Comment:
Why is a second client session created here instead of using the one created previously?